agent/php_user_instrument.c

```diff
@@ -343,6 +363,38 @@ bool nr_php_wraprec_matches(nruserfn_t* p, zend_function* func) {
   if ((NULL == func) || (ZEND_USER_FUNCTION != func->type)) {
     return false;
   }
+
+  if (0 != p->lineno) {
+    /*
+     * Lineno is set in the wraprec.  If lineno doesn't match, we can exit without
+     * going on to the funcname/classname pair comparison.
+     * If lineno matches, but the wraprec filename is NULL, it is inconclusive and we
+     * we must do the funcname/classname compare.
+     * If lineno matches, wraprec filename is not NULL, and it matches/doesn't match,
+     * we can exit without doing the funcname/classname compare.
+     */
+    if (p->lineno != nr_php_zend_function_lineno(func)) {
+      return false;
+    } else {
+      /*
+       * lineno matched, let's check the filename
+       */
+      filename = nr_php_function_filename(func);
+
+      /*
+       * If p->filename isn't NULL, we know the comparison is accurate;
+       * otherwise, it's inconclusive even if we have a lineno because it
+       * could be a cli call or evaluated expression that has no filename.
+       */
+      if (NULL != p->filename) {
+        if (0 == nr_strcmp(p->filename, filename)) {
+          return true;
+        }
+        return false;
+      }
+    }
+  }
```

Comment:
we can remove the `else` since the `if` statement returns
```suggestion
    } 
    /*
     * lineno matched, let's check the filename
     */
    filename = nr_php_function_filename(func);

    /*
     * If p->filename isn't NULL, we know the comparison is accurate;
     * otherwise, it's inconclusive even if we have a lineno because it
     * could be a cli call or evaluated expression that has no filename.
     */
    if (NULL != p->filename) {
      if (0 == nr_strcmp(p->filename, filename)) {
        return true;
      }
      return false;
    }
  }
```